service/command_build_info.go

```diff
@@ -0,0 +1,52 @@
+package service
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+)
+
+type componentsOutput struct {
+	BuildInfo  component.BuildInfo
+	Receivers  []config.Type
+	Processors []config.Type
+	Exporters  []config.Type
+	Extensions []config.Type
+}
+
+// newBuildSubCommand constructs a new cobra.Command sub command using the given CollectorSettings.
+func newBuildSubCommand(set CollectorSettings) *cobra.Command {
+	buildCmd := &cobra.Command{
+		Use:   "build-info",
```

Comment:
@jpkrohling @codeboten

What about just "info" or "components"?

Reply:
i'm in favour of `components` but don't have a strong preference here.

Reply:
`components` would be my choice as well